src/chatbot/intro_chatbot/logic_adapters/name_request_logic_adapter.py

```diff
@@ -0,0 +1,129 @@
+    def __init__(self, chatbot, **kwargs):
+        super().__init__(chatbot, **kwargs)
+        self.db = kwargs.get('database_proxy')
+        self.context = kwargs.get('conversation_context')
+        self.confidence = 0
+        self.robot_name_request = False
+        self.name_response = False
+        self.polish_sentence_tokenizer = SentenceFilter()
+        self.name_requests = self.db.get_responses_list_by_tags(tag="robot_name_request")
+
+    def can_process(self, statement):
+        return self.check_if_input_contains_name_request(parse_input(statement)) \
+               or self.context.get_state('is_during_name_processing')
+
+    def process(self, statement, additional_response_selection_parameters=None):
+        self.context.update_state('is_during_name_processing', True)
+
+        response = self.prepare_response(statement)
+
+        confidence = 0 if len(response) == 0 else 0.9
+
+        result = Statement(statement_utils.prepare_statement(response), in_response_to=TypeOfOperation.NAME.value)
+        result.confidence = confidence
+        self.db.add_new_doc_to_collection(Configuration.RESPONSES_COLLECTION.value,
+                                          confidence=result.confidence,
+                                          response=result.text)
+
+        return result
+
+    def prepare_response(self, statement):
+        is_robot_introduced = self.context.get_state('is_robot_introduced')
+        user_name = self.find_name(statement.text.split())
+        if user_name is not None:
+            self.context.update_state('is_user_introduced', True)
+        is_user_introduced = self.context.get_state('is_user_introduced')
+
+        response = []
+
+        if self.context.get_state('is_after_name_processing') \
+                and self.check_if_input_contains_name_request(
+            parse_input(statement)):  # case when user ask for robot name more than once
+            return self.introduce_robot()
+
+        if is_user_introduced is False and is_robot_introduced is False:  # introduce robot and ask for user name
+            self.context.update_state('is_robot_introduced', True)
+            return self.introduce_robot_and_ask_for_user_name()
+
+        if is_user_introduced is False and user_name is None:
+            response.append(self.ask_for_user_name())
+            self.context.update_state('is_user_introduced', True)
+
+        if is_robot_introduced is False:
+            response.append(self.introduce_robot())
+            self.context.update_state('is_robot_introduced', True)
+
+        if is_user_introduced is True \
+                and not self.context.get_state('is_after_name_processing') \
+                and user_name is not None:
+            response.append(self.greet_user(user_name))
+
+        if is_user_introduced is True and is_robot_introduced is True:
+            self.context.update_state('is_after_name_processing', True)
+        return response
```

Comment:
nice if elif will be used ;)

Reply:
the idea was to construct it as a kind of "decorator" which adds something new to response according to `if` conditions, it won't be equivalent with else

Reply:
ok, btw you can use elif without else